server/bin/pbench-index.py

```diff
@@ -102,6 +104,15 @@ def main(options, name):
 
     idxctx = None
     try:
+        # We're going to need the Postgres DB to track dataset state, so setup
+        # DB access. We need to do this before we create the IdxContext, since
+        # that will need the template DB; so we create a PbenchServerConfig and
+        # Logger independently.
+        config = PbenchServerConfig(options.cfg_name)
+        logger = get_pbench_logger(name, config)
+        Database.init_db(config, logger)
```

Comment:
Why not just perform the `init_db()` at the appropriate place in the `IdxContext()` constructor?

Reply:
I think there's already way too much stuff crammed into `IdxCtx` ... but mostly if we start doing this sort of initialization on interior classes we run the risk of having to deal with double-initialization problems like Nikhil ran into in his unit tests. I wanted to avoid that by keeping it at the outer command level, which really ought to own the configuration environment.

(I really think we should be passing the logger and config into `IdxCtx`, too, instead of having it do all that itself; but I didn't want to tackle that on top of everything else.)

Reply:
Yes, but this generates a second config object (I think the logger instance is the same).

This is half-way in between ...

Reply:
True, but neither are all that expensive. I'd rather pull them out of `IdxCtx` than bring this in, if it comes to that, but is this really so bad?

Reply:
See https://github.com/dbutenhof/pbench/pull/1.

Reply:
OK, a PR to a PR...now _that_ was cool.  (\*mind-blown\*)